airbyte-integrations/connectors/source-tidb/src/main/java/io/airbyte/integrations/source/tidb/TiDBSource.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.tidb;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.ImmutableMap;
+import com.mysql.cj.MysqlType;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.db.jdbc.NoOpJdbcStreamingQueryConfiguration;
+import io.airbyte.integrations.base.IntegrationRunner;
+import io.airbyte.integrations.base.Source;
+import io.airbyte.integrations.base.ssh.SshWrappedSource;
+import io.airbyte.integrations.source.jdbc.AbstractJdbcSource;
+import java.util.List;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class TiDBSource extends AbstractJdbcSource<MysqlType> implements Source {
```

Comment:
Please check the implementation of `mysql-strict-encryt` connector. So far I didn't find any change from Mysql that is specifically applied to TiDB.